deps/rabbit/src/rabbit_msg_store.erl

```diff
@@ -518,14 +523,264 @@ read(MsgId,
     case ets:lookup(CurFileCacheEts, MsgId) of
         [] ->
             Defer = fun() -> {server_call(CState, {read, MsgId}), CState} end,
+            %% @todo It's probably a bug if we don't get a positive ref count.
             case index_lookup_positive_ref_count(MsgId, CState) of
                 not_found   -> Defer();
-                MsgLocation -> client_read1(MsgLocation, Defer, CState)
+                MsgLocation -> client_read2(MsgLocation, Defer, CState)
             end;
         [{MsgId, Msg, _CacheRefCount}] ->
             {{ok, Msg}, CState}
     end.
 
+%% @todo read_many
+%% @todo We can only read_many when the backend index is rabbit_msg_store_ets_index
+%%       since we rely on ets:select to get the index entries.
```

Comment:
no we don't anymore so I think the way read_many is done can be changed so that all backends work with it